File: Sources/XMTPiOS/Mls/Installation.swift
//
//  Installation.swift
//
//
//  Created by Naomi Plasterer on 9/25/24.
//

import Foundation
import LibXMTP

public struct Installation {
	var ffiInstallation: FfiInstallation
    
    init(ffiInstallation: FfiInstallation) {
        self.ffiInstallation = ffiInstallation
    }

    public var id: String {
		ffiInstallation.id.toHex
    }
	
	public var createdAt: Date? {
		guard let timestampNs = ffiInstallation.clientTimestampNs else { return nil }
		return Date(timeIntervalSince1970: TimeInterval(timestampNs) / 1_000_000_000)
	}
}

File: sdks/android/jvm/src/main/java/org/xmtp/kotlin/libxmtp/Installation.kt
package org.xmtp.kotlin.libxmtp

import org.xmtp.kotlin.toHex
import uniffi.xmtpv3.FfiInstallation
import java.util.Date

class Installation(
    private val ffiInstallation: FfiInstallation,
) {
    val installationId: String
        get() = ffiInstallation.id.toHex()
    val createdAt: Date?
        get() =
            ffiInstallation.clientTimestampNs?.let {
                Date(it.toLong())
            }
}
